x/auth/keeper/genesis.go

```diff
@@ -20,8 +20,14 @@ func (ak AccountKeeper) InitGenesis(ctx sdk.Context, data types.GenesisState) {
 	}
 	accounts = types.SanitizeGenesisAccounts(accounts)
 
-	for _, a := range accounts {
-		acc := ak.NewAccount(ctx, a)
+	// Set the accounts and make sure the global account number matches the largest account number (even if zero).
+	var lastAccNum *uint64
+	for _, acc := range accounts {
+		accNum := acc.GetAccountNumber()
+		for lastAccNum == nil || *lastAccNum < accNum {
+			n := ak.GetNextAccountNumber(ctx)
```

Comment:
I was scratching my head at this thinking "this doesn't actually do anything". Then I took a look at `GetNextAccountNumber` and realized it's _mutative_. It really should be called `NextAccountNumber` or `IncrAccountNumber` IMO.

Reply:
Yeah, `Get` makes it sound like a read-only operation but it's doing:
```go
bz = ak.cdc.MustMarshal(&gogotypes.UInt64Value{Value: accNumber + 1})
store.Set(types.GlobalAccountNumberKey, bz)
```

Not sure, if we are backporting to previous versions. @tac0turtle ?

Reply:
Well we're def. backporting to 0.47 (I added the label). Older versions I'm not sure. Which ever way, we can accommodate :)

Reply:
agree, lets at a minimum do 0.47. It would be nice to have the api make it more explicit it changes state.

Reply:
Ok so we rename it?

Reply:
Yes.

Reply:
Done ✅